### PR TITLE
Finish the documentation for sd-event.h

### PR DESCRIPTION
**Pull request by poettering**
This completes the set of sd-event documentation.

I'd be thankful of reviewers would focus on the non-prose changes in this PR. If you find typos in the prose, or would like to make other changes, please file a later PR. I'd rather see this PR go in quickly, even if it has a typo or two, as typos generally don't result in package breakage or immediate bugs.

### CODE REVIEW

**Comment by poettering** [comment]
For those who want to read this in a browser. I ran "make doc-sync" with these patches applied, the docs are now here:

http://www.freedesktop.org/software/systemd/man/sd-event.html

File: man/sd_event_add_child.xml [hunk 1]
```diff
@@ -83,42 +86,75 @@
...
+    returns. Child processes for which no child process state change
+    event sources are installed will not be reaped by the event loop
+    implementation.</para>
+
+    <para>If both a child process state change event source and a
+    <constant>SIGCHLD</constant> signal event source is installed in
+    the same event loop, the configured event source priorities decide
+    which event source is dispatched first. If the signal handler is
+    processed first, it should leave the child processes for which
+    child process state change event sources are installed unreaped.</para>
```
Comment (evverx) [inline man/sd_event_add_child.xml:149]: There is no such info in the `sd_event_add_signal` manpage.

File: man/sd_event_new.xml [hunk 2]
```diff
@@ -103,6 +113,17 @@
...
 
+    <para>After allocating an event loop object, add event sources to
+    it with
+    <citerefentry><refentrytitle>sd_event_add_io</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    <citerefentry><refentrytitle>sd_event_add_time</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    <citerefentry><refentrytitle>sd_event_add_signal</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    <citerefentry><refentrytitle>sd_event_add_child</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    or
+    <citerefentry><refentrytitle>sd_event_add_defer</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    and then execute the event loop using
```
Comment (evverx) [inline man/sd_event_new.xml:124]: should we mention `sd_event_add_post` and `sd_event_add_exit` here?

**Comment by evverx** [comment]
oh, I see

> If you find typos in the prose, or would like to make other changes, please file a later PR.

sorry.
Looks great! Thanks!

**Comment by martinpitt** [comment]
The non-doc changes are simple and LGTM, and @evverx apparently already did an initial once-over for the documentation part. Great work on the docs!

File: man/sd_event_new.xml (hunk 2 above)
Comment (poettering) [inline man/sd_event_new.xml:124]: sd_event_add_post() and sd_event_add_exit() are actually on the same man page as sd_event_add_defer(), hence I chose to omit them, given that they are linked anyway...

**Pull request merged**